File: reference/two-pointer/2108-Find-First-Palindromic-String-in-the-Array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        def util_simultaneously(word):
            low, high = 0, len(word) - 1
            while low < high:
                if word[low] != word[high]:
                    return False
                low += 1
                high -= 1
            return True
            
        for word in words:
            if util_simultaneously(word):
                return word
        return ""

File: reference/two-pointer/2108-Find-First-Palindromic-String-in-the-Array.go
func firstPalindrome(words []string) string {
    for _, word := range words {
		if util_Palindrome(word) {
            return word
        }
	}
    return ""
}

func util_Palindrome(word string) bool {
    low, high := 0, len(word) - 1
    for low < high {
        if word[low] != word[high] {
            return false
        }
        low += 1
        high -= 1
    }
    return true
}